### ISSUE TITLE
Checkov crashes when parsing kube_config field

### ISSUE THREAD

**Issue by holgiosalos** [bridgecrewio/checkov#355]
**Describe the bug**
Parsing fails when processing a resource or provider related to a Kubernetes cluster and the following notation is used in some field `kube_config.0.property` notation.

**To Reproduce**
1. Provider defined in a similar way than in this example:
```
provider "helm" {
  version = "1.1.1"
  kubernetes {
    host                   = module.aks_cluster.kube_config.0.host
    client_certificate     = base64decode(module.aks_cluster.kube_config.0.client_certificate)
    client_key             = base64decode(module.aks_cluster.kube_config.0.client_key)
    cluster_ca_certificate = base64decode(module.aks_cluster.kube_config.0.cluster_ca_certificate)
    load_config_file       = false
  }
}

provider "kubernetes" {
  version                = "1.10.0"
  host                   = module.aks_cluster.kube_config.0.host
  client_certificate     = base64decode(module.aks_cluster.kube_config.0.client_certificate)
  client_key             = base64decode(module.aks_cluster.kube_config.0.client_key)
  cluster_ca_certificate = base64decode(module.aks_cluster.kube_config.0.cluster_ca_certificate)
}
```
2. Module defined as
```
module "new_relic" {
  source                            = "s3::https://s3.amazonaws.com/my-artifacts/new-relic-k8s-0.2.5.zip"
  kubernetes_host                   = module.aks_cluster.kube_config.0.host
  kubernetes_client_certificate     = base64decode(module.aks_cluster.kube_config.0.client_certificate)
  kubernetes_client_key             = base64decode(module.aks_cluster.kube_config.0.client_key)
  kubernetes_cluster_ca_certificate = base64decode(module.aks_cluster.kube_config.0.cluster_ca_certificate)
  cluster_name                      = module.naming_conventions.aks_name
  new_relic_license                 = data.vault_generic_secret.new_relic_license.data["license"]
  cluster_ca_bundle_b64             = module.aks_cluster.kube_config.0.cluster_ca_certificate
  module_depends_on                 = [null_resource.delay_aks_deployments]
}
```
3. Run cli command : checkov -d folder
```
DEBUG:checkov.terraform.parser:failed while parsing file /home/holgiosalos/myterraform/myazuremodule/1-main.tf
Traceback (most recent call last):
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/lark/parsers/lalr_parser.py", line 62, in get_action
    return states[state][token.type]
KeyError: 'DECIMAL'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/checkov/terraform/parser.py", line 42, in hcl2
    tf_definition = self._parse_tf_definitions(tf_file)
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/checkov/terraform/parser.py", line 24, in _parse_tf_definitions
    tf_definition = hcl2.load(file)
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/hcl2/api.py", line 9, in load
    return loads(file.read())
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/hcl2/api.py", line 18, in loads
    return hcl2.parse(text + "\n")
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/lark/lark.py", line 311, in parse
    return self.parser.parse(text, start=start)
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/lark/parser_frontends.py", line 89, in parse
    return self._parse(token_stream, start, *[sps] if sps is not NotImplemented else [])
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/lark/parser_frontends.py", line 54, in _parse
    return self.parser.parse(input, start, *args)
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/lark/parsers/lalr_parser.py", line 36, in parse
    return self.parser.parse(*args)
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/lark/parsers/lalr_parser.py", line 86, in parse
    action, arg = get_action(token)
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/lark/parsers/lalr_parser.py", line 65, in get_action
    raise UnexpectedToken(token, expected, state=state)
lark.exceptions.UnexpectedToken: Unexpected token Token(DECIMAL, '0') at line 30, column 61.
Expected one of: 
	* __ANON_3

2020-06-16 18:58:46,157 [MainThread  ] [DEBUG]  failed while parsing file /home/holgiosalos/myterraform/myazuremodule/1-main.tf
Traceback (most recent call last):
  File "/home/holgiosalos/.local/lib/python3.8/site-packages/lark/parsers/lalr_parser.py", line 62, in get_action
    return states[state][token.type]
KeyError: 'DECIMAL'
```

**Expected behavior**
Not to crash when provider or module uses a similar kubernetes configuration

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04
 - Checkov Version: 1.0.380 

**Additional context**

- Full log file: https://gist.github.com/holgiosalos/15dd01e029180ab2e6cdc22adf45474d

- Same problem if I use the latest docker image. The following output is displayed without any stat:

```
       _               _              
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V / 
  \___|_| |_|\___|\___|_|\_\___/ \_/  
                                      
by bridgecrew.io | version: 1.0.380 
```

**Comment by nimrodkor** [comment]
Hi @holgiosalos !

Since checkov does not scan remote modules, only local files, I'm not able to reproduce the error you mentioned with the provided snippets. However, I see the failure is caused on the file `myterraform/myazuremodule/1-main.tf` could you please share a version of that file?

**Comment by holgiosalos** [comment]
Hi @nimrodkor 

Here is the 1-main.tf file: https://gist.github.com/holgiosalos/5d9a41c41fe23022aeb08e9869f1436f
Also, I noticed that the aks.tf also fails.
aks.tf file: https://gist.github.com/holgiosalos/ada0fe846aa7052bf212e738f48ace72
aks-variables.tf file: https://gist.github.com/holgiosalos/1a512116958e7a8362788bb8a2bbd777

**Comment by nimrodkor** [comment]
Hi @holgiosalos !

I checked these files out and was able to reproduce the error. The problem stems from this line:
`host                   = module.aks_cluster.kube_config.0.host`
Is this a way of accessing the object at index 0 in the list `kube_config`? It seems like it isn't a common terraform pattern, and so fails to be parsed as valid terraform. Can you share a sanitized version of the EKS module? mainly around what produces the output `kube_config`

**Comment by holgiosalos** [comment]
Hi @nimrodkor 

Yes, the `.kube_config.0.host` is a way to accessing the object at index 0. In the list `kube_config`.

There is a similar example for the usage of credentials for the Kubernetes provider at the very end of the following page: https://www.terraform.io/docs/providers/azurerm/r/kubernetes_cluster.html

I could not find a sanitized version of the AKS module. However, if I change the `module.aks_cluster.kube_config.0.host` by `module.aks_cluster.kube_config[0].host` that line stops failing in parse process. So, I guess solution is to change any usage of `kube_config.0.property` by `kube_config[0].property`.

However got another error, this time parsing [aks-variables.tf](https://gist.github.com/holgiosalos/1a512116958e7a8362788bb8a2bbd777) file on line 69, it seems that is this portion of code:
```
  type = list(object({
    group : string
    namespace : string
    role : string
  }))
```

Should I open a feature request or bug for the new parsing error?

**Comment by schosterbarak** [comment]
@holgiosalos we are looking for a way to solve this. we'll keep you posted in a day or two

**Comment by tobypeschel** [comment]
Yep, just encountered this problem myself. Took me an embarrassingly long amount of time to realise it was the following two  lines to blame:

`kubernetes_ca_cert = base64decode(google_container_cluster.primary.master_auth.0.cluster_ca_certificate)`

`name      = kubernetes_service_account.vault-auth.metadata.0.name`

But this is definitely a common pattern and found in many examples from the docs and elsewhere.

**Comment by holgiosalos** [comment]
Hello guys, any updates on this one? @schosterbarak @nimrodkor 

Can I help you with something?

**Comment by nimrodkor** [comment]
hi @holgiosalos !
So apparently this is a parsing failure on checkov's HCL2 library, probably in its dependency, `lark_parser`. We are investigating it and hope to reach the solution for this soon!

**Comment by schosterbarak** [comment]
solved on #477
